srcpkgs/mold/template

```diff
@@ -0,0 +1,45 @@
+# Template file for 'mold'
+pkgname=mold
+version=0.9.1
+revision=1
+build_style=gnu-makefile
+hostmakedepends="clang"
+makedepends="mimalloc-devel openssl-devel xxHash-devel tbb-devel zlib-devel"
+short_desc="High performance drop-in replacement for existing Unix linkers"
+maintainer="Artur Sinila <freesoftware@logarithmus.dev>"
+license="AGPL-3.0-or-later"
+homepage="https://github.com/rui314/mold"
+changelog="https://github.com/rui314/mold/releases"
+distfiles="https://github.com/rui314/mold/archive/refs/tags/v${version}.tar.gz"
+checksum=02b156de6cd2d94fea4eed9748a7c96955673d810ec672359f603f2f90e2990d
+_commit_hash=9a09c777d9460ebe7eb498d1cc0327915f8bbcdc
+
+# use system-wide installed mimalloc instead of the vendored one
+export SYSTEM_MIMALLOC=1
+
+pre_build() {
+	# gcc 10.2 doesn't fully support C++20
+ 	export CXX=clang++
```

Comment:
clang doesn't support C++20 fully either, so this doesn't explain why it's necessary. Using a different compiler for something as transient as standard support isn't great.

Reply:
@ericonr hmm we still have gcc 10.2 in Void, and it's older than clang 12, and that's why our gcc doesn't support c++20. When our gcc will be updated, we will ditch this line.

Reply:
That's not how things work. Compilers move at difference paces, so having something newer or older than the other doesn't really mean a lot. What errors does it get when built with gcc?

Reply:
@ericonr hmm now it compiled just fine... Maybe a change in some of the patches did the trick. Because at the very beginning there were some errors in g++ which I unfortunately don't remember now.